server/src/main/java/org/opensearch/index/query/RangeQueryBuilder.java

```diff
@@ -451,6 +451,11 @@ public String getWriteableName() {
                 return MappedFieldType.Relation.INTERSECTS;
             }
 
+            // For validation, always assume that there is an intersection
+            if (shardContext.validate()) {
```

Comment:
Are we validating the range is in valid DateTimeFormat (can be parsed as such) before bypassing the validation to the other part of the query with INTERSECT? I could be wrong, but seems like the dateMathParser below does some validation for the date format [here](https://github.com/opensearch-project/OpenSearch/blob/996d33adb22ac6a523962166f3677f8dc6ac9c1f/server/src/main/java/org/opensearch/common/time/JavaDateMathParser.java#L245). In case of invalid range, the query should be marked invalid. Couldn't find a test case for this scenario or maybe I missed? Just want to make sure we are not ignoring the case where timestamp is invalid.